documentation/kubernetes-upgrade.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Upgrading kubeadm clusters from v1.7/v1.8 to v1.9 with vSphere Cloud Provider
+---
+
+## Before you begin
+
+Before proceeding:
+
+* You need to have a functional Kubernetes cluster deploed using kubeadm running version 1.7.0 or higher with vSphere Cloud Provider configured in order to use the process described here. If vSphere Cloud Provider in not enabled in your Kubernetes cluster, you can follow the instructions mentioned in this [link] (https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html#manual-steps-if-needed-to-enable-vsphere-cloud-provider) to manually enable vSphere Cloud Provider.
+* Make sure swap is disabled.
+* kubeadm upgrade allows you to upgrade etcd. kubeadm upgrade will also upgrade etcd to 3.1.10 as part of upgrading from v1.8 to v1.9 by default. This is due to the fact that etcd 3.1.10 is the officially validated etcd version for Kubernetes v1.9. The upgrade is handled automatically by kubeadm for you.
+* Note that kubeadm upgrade will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up what’s important to you. For example, any app-level state, such as a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
+
+## Upgrading your control plane
+* Install the most recent version of kubeadm as mentioned in this [link] (https://kubernetes.io/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/)
+
+* On the master node, run the following
+
+        kubeadm upgrade plan
+  This checks that your cluster is upgradeable and fetches the versions available to upgrade to in an user-friendly way.
+
+* Now, pick a version to upgrade to and run.
+
+        kubeadm upgrade apply v1.9.2
+  You can refer the [link] (https://kubernetes.io/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) for detailed instructions.
+
+* Now that your control plane is upgraded, you need to upgrade your kubelets in turn.
+
+## Upgrading your master and node packages
+* For each host (referred to as $HOST below) in your cluster, upgrade kubelet by executing the following commands:
+
+        kubectl drain $HOST --ignore-daemonsets
+
+* Upgrade the Kubernetes package versions on the $HOST node by using a Linux distribution-specific package manager. You can refer this [link] for distro specific instructions.
+
+## Set up vSphere cloud provider to work for a Kubernetes cluster deployed in multi-VC environment during upgrade
+
+The upgrade instructions from Kubernetes 1.7/1.8 to 1.9 using kubeadm as described above works fine for a Kubernetes cluster deployed on a single vCenter.
+
+In 1.9, vSphere Cloud Provider has support for a single Kubernetes cluster deployed across multiple vCenters. So, if one needs to upgrade from Kubernetes cluster 1.7/1.8 deployed on single VC to Kubernetes cluster 1.9 deployed across multiple vCenters, few additiona steps need to be performed apart from those mentioned above.
```

Comment:
First let us talk about upgrading k8s deployed in a single VC. This should be captured in documentation. Its rare that the customer wants to upgrade and spread nodes across VCs in one go.  Even if the customer wants to, this can be in a separate section in the documentation.

Reply:
Added a new header for this in the page.